spec/requests/api/tag_collections_spec.rb

```diff
@@ -397,4 +397,54 @@ def app
       expect_resource_has_tags(service_template, tag2[:path])
     end
   end
+
+  context "Tenant Tag subcollection" do
+    let(:tenant)          { FactoryGirl.create(:tenant, :name => "Tenant A", :description => "Tenant A Description") }
+    let(:tenant_url)      { tenants_url(tenant.id) }
+    let(:tenant_tags_url) { "#{tenant_url}/tags" }
+
+    it "query all tags of a Tenant and verify tag category and names" do
+      api_basic_authorize
+      classify_resource(tenant)
+
+      run_get tenant_tags_url, :expand => "resources"
+
+      expect_query_result(:tags, 2, :tag_count)
+      expect_result_resources_to_include_data("resources", "name" => :tag_paths)
+    end
+
+    it "assigns a tag to a Tenant without appropriate role" do
```

Comment:
The spec description I found confusing here. Is it supposed to say it "does not assign [...]"?